packages/@react-aria/menu/src/useMenu.ts

```diff
@@ -72,11 +75,31 @@ export function useMenu<T>(props: AriaMenuOptions<T>, state: TreeState<T>, ref:
     onAction: props.onAction
   });
 
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      // TODO: Let Tab propagate so FocusScope handle it. Revisit if we decide to close all menus
+      // Also let Escape propagate so useOverlay handles closing the Menu overlay
+      if (e.key === 'Tab' || e.key === 'Escape') {
+        e.continuePropagation();
+      }
+
+      onKeyDown && onKeyDown(e);
+    },
+    onKeyUp(e) {
+      // Need to continue keyup propagation so MenuTrigger button's press state is properly reset, otherwise it gets stuck in pressed state
+      e.continuePropagation();
+      onKeyUp && onKeyUp(e);
+    }
+  });
+
   return {
-    menuProps: mergeProps(domProps, {
+    menuProps: mergeProps(domProps, keyboardProps, {
       role: 'menu',
+      // TODO: we no longer update treestate's expandedKeys so we can't do the below anymore. This affects Safari VO specifically, the VO cursor doesn't move into the newly opened
```

Comment:
yep this is still a TODO mainly for the team to discuss how bad the behavior is and how we'd like to mitigate it. Previously we were setting `aria-hidden` on a menu/submenu once it had an `expandedKey` to force the VO cursor to move to the newly opened submenu. This currently would require us to update each menu's treeState's `expandedKeys` which was something we wanted to move away from in favor of tracking a list of expanded keys in [useMenuTriggerState](https://github.com/adobe/react-spectrum/blob/5da908c1cbbfd868af749705a3cc4c81d6b7b58c/packages/%40react-stately/menu/src/useMenuTriggerState.ts#L51). Note that `state` here is the menu's TreeState rather than the state returned by `useMenuTriggerState`

For now I've gotten rid of the `aria-hidden` here since we deemed it a bit of a edge case in the meeting we had on 9/25, but I could bring it back without needing to use `state.expandedKeys.size` if I add another prop to useMenu for `level`/`hasOpenSubMenu` or if I pass the `subMenuTriggerState` to `useMenu` so I can look up `level` instead